albatross/models/least_squares.h

```diff
@@ -36,22 +30,22 @@ struct LeastSquaresFit {
 };
 
 /*
- * This model supports a family of RegressionModels which consist of
+ * This model supports a family of models which consist of
  * first creating a design matrix, A, then solving least squares.  Ie,
  *
  *   min_x |y - Ax|_2^2
  *
  * The FeatureType in this case is a single row from the design matrix.
  */
-class LeastSquaresRegression
-    : public SerializableRegressionModel<Eigen::VectorXd, LeastSquaresFit> {
+template <typename ImplType>
+class LeastSquares : public ModelBase<LeastSquares<ImplType>> {
```

Comment:
The `ImplType` here is a way to extend `LeastSquares`.  In the example of `LinearRegression` it just provides `fit` and `predict` methods that work with `double` types instead of `Eigen::VectorXd` types, but could for example include things like routines for converting weigthed least squares problems into regular least squares etc ...